test/gradcheck.jl

```diff
@@ -1403,6 +1403,17 @@ end
 end
 
 @testset "AbstractFFTs" begin
+
+  # Many of these tests check a complex gradient to a function with real input. This is now
+  # clamped to real by ProjectTo, but to run the old tests, use here the old gradient function:
+  function oldgradient(f, args...)
+    y, back = Zygote.pullback(f, args...)
+    back(Zygote.sensitivity(y))
+  end
+  # Eventually these rules and tests will be moved to ChainRules.jl, at which point the tests
+  # can be updated to use real / complex consistently.
+  # https://github.com/JuliaMath/AbstractFFTs.jl/pull/58
```

Comment:
Normally I would not be OK with this defining `oldgradient` as it is a super-gross hack.
But if we can outright just delete all these tests entirely once that PR is merge than I am down with it.

Reply:
Yea I lost quite a few hours trying to fix these. They are pretty convoluted tests.